tests/roles/dataplane_adoption/tasks/main.yaml

```
@@ -305,6 +305,70 @@
       }]'
   when: edpm_neutron_dhcp_agent_enabled|bool
 
+- name: Run pre-adoption validation
+  when: run_pre_adoption_validation|bool
+  block:
+    - name: Create OpenStackDataPlaneService/pre-adoption-validation
+      no_log: "{{ use_no_log }}"
+      ansible.builtin.shell: |
+        {{ shell_header }}
+        {{ oc_header }}
+        oc apply -f - <<EOF
+        apiVersion: dataplane.openstack.org/v1beta1
+        kind: OpenStackDataPlaneService
+        metadata:
+          name: pre-adoption-validation
+        spec:
+          playbook: osp.edpm.pre_adoption_validation
+        EOF
+
+    - name: Create OpenStackDataPlaneDeployment to run the validation only
+      no_log: "{{ use_no_log }}"
+      ansible.builtin.shell: |
+        {{ shell_header }}
+        {{ oc_header }}
+        oc apply -f - <<EOF
+        apiVersion: dataplane.openstack.org/v1beta1
+        kind: OpenStackDataPlaneDeployment
+        metadata:
+          name: openstack-pre-adoption
+        spec:
+          nodeSets:
+          - openstack
+          servicesOverride:
+          - pre-adoption-validation
+        EOF
+
+    - name: Wait for the validation deployment to finish
+      ansible.builtin.shell: |
+        {{ shell_header }}
+        {{ oc_header }}
+
+        DEPLOYMENT_NAME=openstack-pre-adoption
+        TRIES=180
+        DELAY=10
+        ALLOWED_JOB_RETRIES=1
+
+        for i in $(seq $TRIES)
+        do
+            ready=$(oc get osdpd/$DEPLOYMENT_NAME -o jsonpath='{.status.conditions[0].status}')
+            if [ "$ready" == "True" ]; then
+                echo "Pre adoption validation Deployment is Ready"
+                exit 0
+            else
+                failed=$(oc get jobs -l osdpd=$DEPLOYMENT_NAME -o jsonpath="{.items[?(@.status.failed > $ALLOWED_JOB_RETRIES)].metadata.name}")
```

osdpd label doesn't exist anymore: https://github.com/openstack-k8s-operators/dataplane-operator/blob/main/pkg/util/ansible_execution.go#L285-L289
```suggestion
                failed=$(oc get jobs -l openstackdataplanedeployment=$DEPLOYMENT_NAME -o jsonpath="{.items[?(@.status.failed > $ALLOWED_JOB_RETRIES)].metadata.name}")
```

---

thanks for noticing. Fixed it.